Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp

```diff
@@ -312,6 +312,9 @@ static GstFlowReturn transformInPlace(GstBaseTransform* base, GstBuffer* buffer)
             GST_ELEMENT_ERROR(self, STREAM, FAILED, ("There is no subsamples buffer, but a positive subsample count"), (nullptr));
             return GST_FLOW_NOT_SUPPORTED;
         }
+    } else if (isCbcs) {
```

Comment:
Could you back this change with some specific text from the ISO/IEC 23001-7 spec?

I'm asking because, when trying to find some documentation to review this change, I found this code in the ffmpeg implementation: https://ffmpeg.org/doxygen/trunk/mov_8c_source.html (line 8023). There, what ffmpeg does is to consider that the whole packet should be decrypted, instead of being left as it is.

I'm not saying that your approach is wrong, but would like to have some rationale that supports it.

Reply:
To be honest, I'm not sure myself if this approach is correct.
I do know that the samples in those specific TCs are clear, unencrypted - because otherwise after this change audio output would be garbage, or the content wouldn't play at all (and it didn't play before the fix on our boxes).

I've noticed that Cobalt, which is also using ocdm library, didn't call ocdm decryption functions in those TCs. That's how I noticed this content is actually clear.
BTW it's strange - what's the point of having AC3 **cbcs** testcase where all of the samples are clear? (yes, all of them - for comparison, in AACAudioCbcs TC, some samples are encrypted, some are not). So it might also be a bug in YouTube's testcases.

I'll try to go through the spec (if I manage to get access to it) and verify it

Reply:
Well... The Cobalt counterexample is a good one. Maybe it's easier to try to find the code there that justifies that behaviour.
Still, it would be nice to know @calvaris' opinion about this topic, as he has more experience with CBCS than me.